### PR TITLE
feature: add RESTMiniChart HOC

### LINKED ISSUE #4418: feature: add RESTMiniChart HOC
# Feature Request

## User Story
As a developer, I want a higher order component that handles fetching data from a REST Endpoint and passing it to a MiniChart so that the MiniChart can remain solely presentational. The RESTMiniChart component should accept a REST route, and title.

## Possible Solution
Use axios to fetch data from the REST API with the given route. Then, pass that data to a child MiniChart component.

## Acceptance Criteria
- [ ] Data is fetched from given URL and passed to child MiniChart on page load
- [ ] Data is fetched from given URL and passed to child MiniChart on change in period selection
- [ ] Component uses internal state to manage loading states

### PR DESCRIPTION
**Pull request by henryholtgeerts**
## Description
Resolves #4418 
This PR adds the RESTMiniChart HOC, which handles fetching data from the Reports API and passes it to a child MiniChart component. The PR also moves logic for finding averages, totals and trends from the frontend to the API endpoint, to take advantage of existing currency formatting that exists in the plugin. 

## How Has This Been Tested?
Has been tested locally and throws no errors in the browser console. Passed PHPUnit tests.

## Types of changes
New feature (non-breaking change which adds functionality)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.

### CODE REVIEW

File: assets/src/js/admin/reports/app/index.js [hunk 1]
```diff
@@ -1,42 +1,39 @@
...
-            </div>
-        </StoreProvider>
-    )
-}
-export default App
+	return (
+		<StoreProvider initialState={ initialState } reducer={ reducer }>
+			<div className="wrap give-settings-page">
+				<div className="give-settings-header">
+					<h1 className="wp-heading-inline">Reports</h1>
```
Comment (ravinderk) [inline assets/src/js/admin/reports/app/index.js]: @henryholtgeerts can we localize this string?

File: assets/src/js/admin/reports/components/mini-chart/index.js [hunk 2]
```diff
@@ -1,81 +1,85 @@
...
-                </div>
-            </div>
-        </div>
-    )
-}
+	return (
+		<div className="givewp-mini-chart">
+			<div className="header">
+				<div className="title">{ title }</div>
+				<div className="indicator">{ indicator && ( indicator ) }</div>
```
Comment (ravinderk) [inline assets/src/js/admin/reports/components/mini-chart/index.js]: @henryholtgeerts what `{ indicator && ( indicator ) }` and `{ highlightValue && ( highlightValue ) }` this expression does

Comment (henryholtgeerts) [reply assets/src/js/admin/reports/components/mini-chart/index.js]: Here I am checking that these variables have been defined before attempting to render them. https://reactjs.org/docs/conditional-rendering.html#inline-if-with-logical--operator

File: assets/src/js/admin/reports/components/mini-chart/index.js [hunk 3]
```diff
@@ -1,81 +1,88 @@
...
-    const canvas = createRef()
-    const config = createConfig(data)
+	const canvas = createRef();
+	const config = createConfig( data );
 
-    useEffect(() => {
+	useEffect( () => {
+		const newHighlightValue = getHighlightValue( highlight, data );
+		const newTrend = getTrend( data );
+		const newIndicator = Math.sign( trend ) === -1 ? <Fragment>{ down } <span style={ { color: '#D75A4B' } }>{ Math.abs( trend ) + '%' }</span></Fragment> : <Fragment>{ up } <span style={ { color: '#69B868' } }>{ Math.abs( trend ) + '%' }</span></Fragment>;
```
[the remark this answers is not in the document]
Comment (henryholtgeerts) [reply assets/src/js/admin/reports/components/mini-chart/index.js]: Changes made, ref: https://github.com/impress-org/givewp/pull/4424/commits/33a94185f621fb8cd84f58d9e54404203a43272e

File: assets/src/js/admin/reports/app/index.js (hunk 1 above)
Comment (henryholtgeerts) [reply assets/src/js/admin/reports/app/index.js]: Change made, ref: https://github.com/impress-org/givewp/pull/4424/commits/97c63ca856fad2bddfb3497be85d6a5bd3a31e0e

**Pull request merged**